### ISSUE TITLE
@vue/cli-plugin-eslint - Other ESLint CLI options are *not* also supported.

### ISSUE THREAD

**Issue by DaedalusDev** [vuejs/vue-cli#4007]
### Version
3.7.0

### Environment info
```
Environment Info:

  System:
    OS: Linux 4.4 Ubuntu 16.04.6 LTS (Xenial Xerus)
    CPU: (8) x64 Intel(R) Xeon(R) CPU E5-1620 v3 @ 3.50GHz
  Binaries:
    Node: 8.12.0 - ~/node-current/bin/node
    Yarn: 1.3.2 - ~/node-current/bin/yarn
    npm: 6.9.0 - ~/node-current/bin/npm
  Browsers:
    Chrome: 73.0.3683.75
    Firefox: 60.2.0
  npmPackages:
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0 
    @vue/babel-plugin-transform-vue-jsx:  1.0.0 
    @vue/babel-preset-app:  3.7.0 
    @vue/babel-preset-jsx:  1.0.0 
    @vue/babel-sugar-functional-vue:  1.0.0 
    @vue/babel-sugar-inject-h:  1.0.0 
    @vue/babel-sugar-v-model:  1.0.0 
    @vue/babel-sugar-v-on:  1.0.0 
    @vue/cli-overlay:  3.7.0 
    @vue/cli-plugin-babel: ^3.7.0 => 3.7.0 
    @vue/cli-plugin-e2e-cypress: ^3.7.0 => 3.7.0 
    @vue/cli-plugin-eslint: ^3.7.0 => 3.7.0 
    @vue/cli-plugin-pwa: ^3.7.0 => 3.7.0 
    @vue/cli-plugin-unit-jest: ^3.7.0 => 3.7.0 
    @vue/cli-service: ^3.7.0 => 3.7.0 
    @vue/cli-shared-utils:  3.7.0 
    @vue/component-compiler-utils:  2.6.0 
    @vue/devtools: ^1.1.0 => 1.1.0 
    @vue/eslint-config-standard: ^4.0.0 => 4.0.0 
    @vue/preload-webpack-plugin:  1.1.0 
    @vue/test-utils: ^1.0.0-beta.29 => 1.0.0-beta.29 
    @vue/web-component-wrapper:  1.2.0 
    eslint-plugin-vue:  4.7.1 
    jest-serializer-vue:  2.0.2 
    vue: ^2.6.10 => 2.6.10 
    vue-cli-plugin-quasar: ^1.0.0-beta.1 => 1.0.0-beta.1 
    vue-d3-network: ^0.1.109 => 0.1.109 
    vue-eslint-parser:  2.0.3 
    vue-hot-reload-api:  2.3.3 
    vue-jest:  3.0.4 
    vue-loader:  15.7.0 
    vue-masonry-css: ^1.0.2 => 1.0.3 
    vue-router: ^3.0.2 => 3.0.2 
    vue-server-renderer:  2.6.10 
    vue-split-panel: ^1.0.4 => 1.0.4 
    vue-style-loader:  4.1.2 
    vue-template-compiler: ^2.6.10 => 2.6.10 
    vue-template-es2015-compiler:  1.9.1 
    vuedraggable: ^2.20.0 => 2.20.0 
    vuelidate: ^0.7.4 => 0.7.4 
    vuex: ^3.1.0 => 3.1.0 
  npmGlobalPackages:
    @vue/cli: 3.5.5

```

### Steps to reproduce
run `vue-cli-service lint` with extra options (ex: `node ./node_modules/.bin/vue-cli-service lint -o report.json -f json`)

### What is expected?
Extra params would be proxied to eslint cli

### What is actually happening?
No extra options are supported.

**Comment by LinusBorg** [comment]
This is mostly a documentation issue. 

Since we don't run the eslint *cli*, but instead use its Node API, we in dvd only support options that are not related to the eslint CLI output. 

So you can pass options like `--exclude`, but not `-o` or `-f`

**Comment by cjackson** [comment]
We can specify `--format` on the `vue-cli-service lint` command. Is there a way to output the result to a file, similar to the `-o` option on the eslint CLI?

**Comment by LinusBorg** [comment]
No.

**Comment by DaedalusDev** [comment]
@cjackson You can simply use stdout syntax like `node ./node_modules/.bin/vue-cli-service lint --format json > report.json` but unfortunally, if all is clean, the command will output something line ` DONE  No lint errors found!`...

Maybe a feature request ?

**Comment by cjackson** [comment]
@DaedalusDev That's what I ended up doing using a bash script to clean up the output a bit. I'll make a feature request for this later today or tomorrow.

**Comment by chrisbraddock** [comment]
This would be nice for collecting [linting metadata in CI](https://circleci.com/docs/2.0/collect-test-data/#eslint).

**Comment by hskrishna29** [comment]
> @DaedalusDev That's what I ended up doing using a bash script to clean up the output a bit. I'll make a feature request for this later today or tomorrow.

I ran into this issue when I wanted to output the lint results to a file. Found out its not supported. Any advancements with respect to this?

**Comment by zwik** [comment]
I'm also interested in this, I would love to push the ESLint report into SonarQube because the linting in SonarQube is not so configurable as ESLint is.

As far as I understand now, it appears there's no work around yet? At least with the suggestions above I'm not able to get a usable file.

**Comment by LinusBorg** [comment]
> As far as I understand now, it appears there's no work around yet

I would say that running eslint directly is a totally fine workaroud. Even though [lint.js](https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-plugin-eslint/lint.js) seems to do a lot of stuff, most of that is about either
* loading the config (which eslint cli does for you)
* configuring the CLIEngine (which you also don't have to when running eslint via cli directly)
* setting some default folders and file types to lint

The latter can be done quickly by adding your own globs when running eslint via cli.

So in summary: **The `lint` command itself doesn't do much more than run eslint**

Given that, and given that there are eslint options that only work with eslint CLI, not with eslint CLIEngine, I don't think that it's worth re-creating these options in our codebase, as it doesn't take much to run eslint "manually" and use the options directly.

I think it would make more sense to document how to run eslint manually (with which globs etc) to make it work exactly like `lint` - then people can use that when they need to use eslint options that only work in the cli version.

**Comment by probil** [comment]
Thanks @LinusBorg I fixed it this way

```json
// package.json
"lint": "npx eslint src test *.js"
```

Then during CI:
```bash
npm run lint -- --format junit -o /tmp/test-results/code-style-test-results.xml
```

**Comment by cawoodm** [comment]
> I would say that running eslint directly is a totally fine workaroud. 
I thought the whole point of this plugin is that it makes ESLint work for .vue files? How is ESLint going to lint .vue files on it's own?
